heron/schedulers/src/java/com/twitter/heron/scheduler/yarn/HeronMasterDriver.java

```diff
@@ -139,107 +142,129 @@ public HeronMasterDriver(EvaluatorRequestor requestor,
   /**
    * Requests container for TMaster as container/executor id 0.
    */
-  void scheduleTMasterContainer() {
+  boolean scheduleTMasterContainer() {
     LOG.log(Level.INFO, "Scheduling container for TM: {0}", topologyName);
-    try {
-      // TODO: https://github.com/twitter/heron/issues/951: colocate TMaster and Scheduler
-      launchContainerForExecutor(TMASTER_CONTAINER_ID, 1, TM_MEM_SIZE_MB);
-    } catch (InterruptedException e) {
-      // Deployment of topology fails if there is a error starting TMaster
-      throw new RuntimeException("Error while waiting for topology master container allocation", e);
-    }
+    // TODO: https://github.com/twitter/heron/issues/951: colocate TMaster and Scheduler
+    AllocatedEvaluator container =
+        launchContainerForExecutor(new HeronWorker(TMASTER_CONTAINER_ID, 1, TM_MEM_SIZE_MB));
+    return container != null;
   }
 
   /**
    * Container allocation is asynchronous. Request containers serially to ensure allocated resources
    * match the required resources
    */
-  void scheduleHeronWorkers(PackingPlan topologyPacking) {
+  boolean scheduleHeronWorkers(PackingPlan topologyPacking) {
```

Comment:
A HeronWorker represents a YARN container for heron instances. Scheduler and TMaster containers are managed differently.